### DISCUSSION TITLE
Salt can't be installed from source: can someone with a cleaner brew install help me reproduce this bug?

### DISCUSSION THREAD

**Discussion by ScoreUnder** [Homebrew/discussions#4442] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.14-74-g71de574
ORIGIN: https://github.com/Homebrew/brew
HEAD: 71de57428568292bd3f5aa52250dd4eb39c3453e
Last commit: 10 hours ago
Core tap JSON: 20 Apr 00:56 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.32.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3-arm64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 14.2.


Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libazureObjectAccess.dylib
  /usr/local/lib/libazurestorage.7.5.dylib
  /usr/local/lib/libcpprest.2.10.dylib
  /usr/local/lib/libcrypto.1.0.0.dylib
  /usr/local/lib/libicudataAvid.dylib
  /usr/local/lib/libicuucAvid.dylib
  /usr/local/lib/libssl.1.0.0.dylib
  /usr/local/lib/libuuid.16.dylib

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  salt@3004.2
```

### Description of issue

Note that the system described has various issues like system update failing silently (or with a nondescript error message and no more verbose output, when invoked from the command line), so many issues in `brew doctor` are currently prohibitively difficult for me to solve and right now I am not particularly interested in solving them.

I would like someone with a cleaner environment and supported install of brew to help reproduce and report a bug for me: the `salt` formula cannot be installed from source (`brew install -s salt`) on an up-to-date system with the latest package upgrades from brew, because `libgit2` is too new and `pygit2` in the `salt.rb` file is too old.

My proposed solution is to update pygit2 to 1.12.0:
```diff
   resource "pygit2" do
-    url "https://files.pythonhosted.org/packages/e7/8a/e52a1c8b9878e9d9743089393f8289bb9c8a81eaab722df22df46a38b9e9/pygit2-1.10.0.tar.gz"
-    sha256 "7c751eee88c731b922e4e487ee287e2e40906b2bd32d0bfd2105947f63e867de"
+    url "https://files.pythonhosted.org/packages/7f/00/075f21ae474fcef679ba1f71b9ecd534493792b508b1919021fb2be67eba/pygit2-1.12.0.tar.gz"
+    sha256 "e9440d08665e35278989939590a53f37a938eada4f9446844930aa2ee30d73be"
   end
```

Thanks all!

**Comment by carlocab** [comment]
Why are you trying to build it from source?

**Reply by ScoreUnder** [reply]
I'm trying to pin the version to match other machines in the fleet, and it has recently stopped building coinciding with a new OS version requirement, and this build failure traces back to the official package.

However, its build failure also implies that the binaries are no longer appropriate to use with the new libgit2 version; they are compiled against an old version and do not want to compile against the newer version.

**Reply by carlocab** [reply, marked as answer]
I see. Updating the `pygit2` resource sounds like it'll work. We'll review a pull request that updates that.

**Reply by ScoreUnder** [reply]
Thank you! I'll put one through in a sec

**Reply by ScoreUnder** [reply]
Okay, I need a little help: how do I `brew edit` the formula from core? I want to be able to test the exact change I make properly before I PR. It tells me the formula isn't located on disk, and I can't seem to find a way to get it to cache it. I could test in a temporary tap but that seems like the wrong way to go about it.

The official docs suggest to do this:
```
$ cd "$(brew --repository homebrew/core)"
-bash: cd: /opt/homebrew/Library/Taps/homebrew/homebrew-core: No such file or directory
```
But that isn't on the disk either, apparently :/

ed: just tapping the fork and hoping it works. I hope bumping the revision number manually is the right option here.

ed2:

`brew test` doesn't work in a tapped fork:
```
[...]
Bundle complete! 37 Gemfile dependencies, 80 gems now installed.
Bundled gems are installed into `../../../../Homebrew/vendor/bundle`
Error: Formulae found in multiple taps: 
       * scoreunder/core/openssl@1.1
       * scivisum/tap/openssl@1.1

Please use the fully-qualified name (e.g. scoreunder/core/openssl@1.1) to refer to the formula.
```

`brew audit` shows an error that I'm not sure is relevant:
```
scoreunder/core:
  * audit_exceptions/versioned_formula_dependent_conflicts_allowlist.json references
    formulae or casks that are not found in the scoreunder/core tap.
    Invalid formulae or casks: python
Error: 1 problem in 1 tap detected
```

still not sure if I should be editing the revision myself or if a bot knows to do that (I'm assuming a bot does all the bottle stuff)

oh well! i'll put the PR through anyway

**Reply by carlocab** [reply]
For future reference, you could've started with
```
brew tap homebrew/core
```
and then follow the docs. They should probably already say that, but they're a bit outdated, unfortunately.
